server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java

```diff
@@ -174,25 +172,21 @@ protected void ensureMaxSeqNoEqualsToGlobalCheckpoint(final SeqNoStats seqNoStat
         if (requireCompleteHistory == false) {
             return;
         }
-        // Before 8.0 the global checkpoint is not known and up to date when the engine is created after
+        // Before 3.0 the global checkpoint is not known and up to date when the engine is created after
         // peer recovery, so we only check the max seq no / global checkpoint coherency when the global
         // checkpoint is different from the unassigned sequence number value.
         // In addition to that we only execute the check if the index the engine belongs to has been
         // created after the refactoring of the Close Index API and its TransportVerifyShardBeforeCloseAction
         // that guarantee that all operations have been flushed to Lucene.
-        final Version indexVersionCreated = engineConfig.getIndexSettings().getIndexVersionCreated();
-        if (indexVersionCreated.onOrAfter(LegacyESVersion.V_7_2_0)
-            || (seqNoStats.getGlobalCheckpoint() != SequenceNumbers.UNASSIGNED_SEQ_NO)) {
-            assert assertMaxSeqNoEqualsToGlobalCheckpoint(seqNoStats.getMaxSeqNo(), seqNoStats.getGlobalCheckpoint());
-            if (seqNoStats.getMaxSeqNo() != seqNoStats.getGlobalCheckpoint()) {
-                throw new IllegalStateException(
-                    "Maximum sequence number ["
-                        + seqNoStats.getMaxSeqNo()
-                        + "] from last commit does not match global checkpoint ["
-                        + seqNoStats.getGlobalCheckpoint()
-                        + "]"
-                );
-            }
+        assert assertMaxSeqNoEqualsToGlobalCheckpoint(seqNoStats.getMaxSeqNo(), seqNoStats.getGlobalCheckpoint());
```

Comment:
Seems like this condition is gone `(seqNoStats.getGlobalCheckpoint() != SequenceNumbers.UNASSIGNED_SEQ_NO)` , on purpose?

Reply:
Correct.. In all modern versions this logic block (checking that the max sequence number is always equal to the global checkpoint) is always executed.  Prior to 7.2 it was only checked if the global checkpoint was was unassigned.